File: src/main/java/com/schoolproject/airbnbclone/models/Location.java
package com.schoolproject.airbnbclone.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Location {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Long id;

    @Column(nullable = false)
    private Double longitude;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private String address;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false)
    private String state;

    @Column(nullable = false)
    private String zipcode;
}

File: src/models/Location.ts
export interface ILocation {
    longitude: number;
    latitude: number;
    address: string;
    city: string;
    state: string;
    zipcode: string;
}

export class Location implements ILocation {
    longitude: number;
    latitude: number;
    address: string;
    city: string;
    state: string;
    zipcode: string;

    constructor(
        longitude: number,
        latitude: number,
        address: string,
        city: string,
        state: string,
        zipcode: string,
    ) {
        this.longitude = longitude;
        this.latitude = latitude;
        this.address = address;
        this.city = city;
        this.state = state;
        this.zipcode = zipcode;
    }
}